### PR TITLE
[INLONG-3473][CI] Add support for chart testing

### PR DESCRIPTION
**Pull request by shink**
Fixes: #3473

### Motivation

Use [helm/chart-testing-action](https://github.com/helm/chart-testing-action) to test helm charts on GitHub Actions.
When the helm charts changes, they can be tested automatically.

### Modifications

- Add chart test workflow

### Documentation

Have updated documentation, please check out https://github.com/shink/incubator-inlong/commit/45c705f611b791538fcd55007a41a6c85742bf2b

### CODE REVIEW

File: .github/kind.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+apiVersion: kind.x-k8s.io/v1alpha4
+name: inlong-cluster
+nodes:
+  - role: control-plane
+  - role: worker
+  - role: worker
+  - role: worker
+  - role: worker
+  - role: worker
+  - role: worker
```
Comment (shink) [inline .github/kind.yml:28]: @dockerzhang Is this OK?

File: .github/workflows/ci_chart_test.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          - 'kindest/node:v1.14.10'
+          - 'kindest/node:v1.15.12'
+          - 'kindest/node:v1.16.15'
+          - 'kindest/node:v1.17.17'
+          - 'kindest/node:v1.18.20'
+          - 'kindest/node:v1.19.16'
+          - 'kindest/node:v1.20.15'
+          - 'kindest/node:v1.21.10'
+          - 'kindest/node:v1.22.7'
+          - 'kindest/node:v1.23.4'
```
Comment (shink) [inline .github/workflows/ci_chart_test.yml:59]: @dockerzhang Is this OK?

**Comment by shink** [comment]
You can see https://github.com/shink/incubator-inlong/actions/workflows/ci_chart_test.yml

File: .github/workflows/ci_chart_test.yml (hunk 2 above)
Comment (dockerzhang) [reply .github/workflows/ci_chart_test.yml:59]: too many versions.

File: .github/kind.yml (hunk 1 above)
Comment (dockerzhang) [reply .github/kind.yml:28]: ok

**Review by healchow — APPROVED** [review]

File: .github/workflows/ci_chart_test.yml (hunk 2 above)
Comment (shink) [reply .github/workflows/ci_chart_test.yml]: Done.

**Comment by shink** [comment]
@dockerzhang Fixed. Please check out https://github.com/apache/incubator-inlong/actions/workflows/ci_chart_test.yml

**Comment by dockerzhang** [comment]
@shink 
1, please rebase after #4121 merged.
2, choose k8s version v1.21+ to avoid `policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+` warnning.

**Comment by shink** [comment]
> 2, choose k8s version v1.21+ to avoid policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+ warnning.

@dockerzhang 
We can compate with different versions of Kubernetes in Helm Chart[^1].
For example:
```
Warning: policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
```
Then we can check `apiVersion` in the Chart.

```yaml
{{- if .Capabilities.APIVersions.Has "policy/v1" }}
apiVersion: policy/v1
{{- else }}
apiVersion: policy/v1beta1
{{- end }}
```

[^1]: https://os.51cto.com/article/689655.html

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**